Mobile/stockdogmobile/screens/login.js

```diff
@@ -8,90 +8,91 @@ import { colors } from '../style/colors';
 import styles from '../style/screens/loginRegister';
 import FormInput from '../components/formInput';
 import WideButton from '../components/widebutton';
-import { loginUser } from '../actions';
-
-mapDispatchToProps = (dispatch) => ({
-  loginUser: (username, password) => {
-    dispatch(loginUser(username, password));
-  }
-});
+import { loginUser } from '../actions/authActions';
+import { login } from '../api';
 
 var logoImage = require('../assets/logo.png');
 
 class Login extends Component {
-  constructor(props) {
-    super(props);
-    var user = "";
-    if (this.props.navigation.state.params) {
-      user = this.props.navigation.state.params.email;
-    }
-    this.state = { 
-      email : user,
-      password: ""
-    };
+   constructor(props) {
+      super(props);
+      var user = this.props.email ? this.props.email : "";
+      this.state = {
+         email: user,
+         password: ""
+      };
+
+      this.inputs = {};
+   }
 
-    this.inputs = {};
-  }
+   focusNextField = (id) => {
+      this.inputs[id].focus();
+   };
 
-  focusNextField = (id) => {
-    this.inputs[id].focus();
-  };
-  
-  navToRegister = () => {
-    Actions.register({});
-  };
+   navToRegister = () => {
+      Actions.register({});
+   };
 
-  login = () => {
-    this.props.loginUser(this.state.email, this.state.password);
-  };
+   submitLogin = () => {
+      login(this.state.email, this.state.password).then((res) => {
+         this.props.loginUser(res.data.userId, res.data.token);
+         Actions.portfolio();
+      }).catch((e) => {
+         console.log(e.response);
```

Comment:
How are you planning to do error messages for the user?

Reply:
Oops I forgot about that good catch. I'll make an alert!

Reply:
Fixed :)